crates/forge/tests/fixtures/zk/Create2.s.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity ^0.8.18;
+
+import {Script} from "forge-std/Script.sol";
+import {Greeter} from "../src/Greeter.sol";
+import {CustomNumber} from "../src/CustomNumber.sol";
+
+contract Create2Script is Script {
+    function run() external {
+        (bool success,) = address(vm).call(abi.encodeWithSignature("zkVm(bool)", true));
+        require(success, "zkVm() call failed");
+        
+        vm.startBroadcast();
+
+        // Deploy Greeter using create2 with a salt
+        bytes32 greeterSalt = bytes32("12345");
+        Greeter greeter = new Greeter{salt: greeterSalt}();
+
+        // Verify Greeter deployment
+        require(address(greeter) != address(0), "Greeter deployment failed");
+
+        // Test Greeter functionality
+        string memory greeting = greeter.greeting("Alice");
+        require(bytes(greeting).length > 0, "Greeter greeting failed");
+
+        // Deploy CustomNumber using create2 with a salt value
+        uint8 customNumberValue = 42;
+        CustomNumber customNumber = new CustomNumber{salt: "123" }(customNumberValue);
```

Comment:
it's not bad to test 2 deployments but at the same time is it necessary?
nit: even in that case, please keep the pattern and syntax consistent: `bytes32` for greeter meanwhile here directly string

Reply:
I think the other tx is with constructor arg so is actually a valid test case.